main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,57 +56,119 @@ public void validate(NoticeContainer noticeContainer) {
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
+          continue;
+        }
+        if (prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addValidationNotice(new DecreasingShapeDistanceNotice(prev, curr));
+          continue;
+        }
+        if (prev.shapeDistTraveled() != curr.shapeDistTraveled()) {
+          continue;
+        }
+        // equal shape_dist_traveled and different coordinates
+        if (!(curr.shapePtLon() == prev.shapePtLon() && curr.shapePtLat() == prev.shapePtLat())) {
           noticeContainer.addValidationNotice(
-              new DecreasingOrEqualShapeDistanceNotice(
-                  curr.shapeId(),
-                  curr.csvRowNumber(),
-                  curr.shapeDistTraveled(),
-                  curr.shapePtSequence(),
-                  prev.csvRowNumber(),
-                  prev.shapeDistTraveled(),
-                  prev.shapePtSequence()));
+              new EqualShapeDistanceDiffCoordinatesNotice(prev, curr));
+        } else {
+          // equal shape_dist_traveled and same coordinates
+          noticeContainer.addValidationNotice(
+              new EqualShapeDistanceSameCoordinatesNotice(prev, curr));
         }
       }
     }
   }
 
   /**
-   * When sorted on `shapes.shape_pt_sequence` key, shape points should have strictly increasing
+   * When sorted on {@code shapes.shape_pt_sequence} key, shape points must have strictly increasing
    * values for `shapes.shape_dist_traveled`
    *
    * <p>"Values must increase along with shape_pt_sequence."
    * (http://gtfs.org/reference/static/#shapestxt)
    *
    * <p>Severity: {@code SeverityLevel.ERROR}
    */
-  static class DecreasingOrEqualShapeDistanceNotice extends ValidationNotice {
+  static class DecreasingShapeDistanceNotice extends ValidationNotice {
+    private final String shapeId;
+    private final long csvRowNumber;
+    private final double shapeDistTraveled;
+    private final int shapePtSequence;
+    private final long prevCsvRowNumber;
+    private final double prevShapeDistTraveled;
+    private final int prevShapePtSequence;
+
+    DecreasingShapeDistanceNotice(GtfsShape current, GtfsShape previous) {
+      super(SeverityLevel.ERROR);
+      this.shapeId = current.shapeId();
+      this.csvRowNumber = current.csvRowNumber();
+      this.shapeDistTraveled = current.shapeDistTraveled();
+      this.shapePtSequence = current.shapePtSequence();
+      this.prevCsvRowNumber = previous.csvRowNumber();
+      this.prevShapeDistTraveled = previous.shapeDistTraveled();
+      this.prevShapePtSequence = previous.shapePtSequence();
+    }
+  }
+
+  /**
+   * When sorted on {@code shapes.shape_pt_sequence} key, close shape points should not have equal
```

Comment:
Since we have removed the threshold, should we keep the "close"?

Reply:
Indeed, let me remove that.